Dockerfile

```diff
@@ -5,42 +5,66 @@ FROM timescale/timescaledb:${TIMESCALEDB_VERSION}-${PG_VERSION_TAG} as builder
 MAINTAINER Timescale https://www.timescale.com
 ARG PG_VERSION_TAG
 
-RUN set -ex \
-    && apk add --no-cache --virtual .build-deps \
-                coreutils \
-                dpkg-dev dpkg \
-                gcc \
-                libc-dev \
-                make \
-                util-linux-dev \
-                clang \
-		llvm \
-                git \
-                llvm-dev clang-libs \
-		bison \
-		dpkg-dev dpkg \
-		flex \
-		gcc \
-		libc-dev \
-		libedit-dev \
-		libxml2-dev \
-		libxslt-dev \
-		linux-headers \
-	 	clang g++ \
-		make \
-		openssl-dev \
-		perl-utils \
-		perl-ipc-run \
-		util-linux-dev \
-		zlib-dev \
-		icu-dev
+RUN set -ex; \
+    apk add --no-cache --virtual .build-deps \
+        coreutils \
+        dpkg-dev dpkg \
+        gcc \
+        libc-dev \
+        make \
+        util-linux-dev \
+        clang \
+        llvm \
+        git \
+        llvm-dev clang-libs \
+        bison \
+        dpkg-dev dpkg \
+        flex \
+        gcc \
+        libc-dev \
+        libedit-dev \
+        libxml2-dev \
+        libxslt-dev \
+        linux-headers \
+        clang g++ \
+        make \
+        openssl-dev \
+        perl-utils \
+        perl-ipc-run \
+        util-linux-dev \
+        zlib-dev \
+        icu-dev
 
-ENV RUSTUP_HOME=/usr/local/rustup \
-    CARGO_HOME=/usr/local/cargo \
-    PATH=/usr/local/cargo/bin:$PATH \
-    RUST_VERSION=1.53.0 \
-    # prevents some issues with openssl
-    RUSTFLAGS="-C target-feature=-crt-static"
+RUN set -ex; \
+    git clone  --branch v1.12 --depth 1 \
+         https://github.com/dimitri/pgextwlist.git /pgextwlist; \
+    cd /pgextwlist; \
+    make; \
+    make install; \
+    mkdir `pg_config --pkglibdir`/plugins; \
+    cp /pgextwlist/pgextwlist.so `pg_config --pkglibdir`/plugins; \
+    rm -rf /pgextwlist
+
+RUN set -eux; \
+    SCCACHE_VERSION=v0.2.15; \
+    APK_ARCH="$(apk --print-arch)"; \
+    SCCACHE_FILE="sccache-${SCCACHE_VERSION}-${APK_ARCH}-unknown-linux-musl" \
+    SCCACHE_ARCHIVE="${SCCACHE_FILE}.tar.gz"; \
+    SCCACHE_URL="https://github.com/mozilla/sccache/releases/download/${SCCACHE_VERSION}/${SCCACHE_ARCHIVE}"; \
+    wget "${SCCACHE_URL}"; \
+    tar -xf ${SCCACHE_ARCHIVE}; \
+    cp ${SCCACHE_FILE}/sccache /usr/local/bin; \
+    chmod +x /usr/local/bin/sccache
+ENV RUSTC_WRAPPER=sccache
+
+USER postgres
+
+WORKDIR /var/lib/postgresql
+
+ENV PATH=/var/lib/postgresql/.cargo/bin:$PATH
+ENV RUST_VERSION=1.57.0
+# prevents some issues with openssl
+ENV RUSTFLAGS="-C target-feature=-crt-static"
```

Comment:
Unfortunately it seems as though removing this breaks things.

Reply:
Could you paste the error you observed or what was broken? That feature is enabled by default on musl targets, so setting this should be a no-op anyway.

Reply:
So the issue that I saw is that after installing `cargo-pgx`, the `cargo pgx init ...` command failed with the following error:
```
could not execute process `/var/lib/postgresql/.cargo/bin/cargo-pgx pgx init --pg14 /usr/local/bin/pg_config` (never executed)
```

more context:

```
#9 141.8   Installing /var/lib/postgresql/.cargo/bin/cargo-pgx
#9 141.8    Installed package `cargo-pgx v0.2.6` (executable `cargo-pgx`)
#9 142.0 + which pg_config
#9 142.0 + cargo pgx init --pg14 /usr/local/bin/pg_config
#9 142.0 error: could not execute process `/var/lib/postgresql/.cargo/bin/cargo-pgx pgx init --pg14 /usr/local/bin/pg_config` (never executed)
#9 142.0
#9 142.0 Caused by:
#9 142.0   No such file or directory (os error 2)```

Reply:
I'm going to pull your branch quick and try to see what the underlying issue is, something is definitely wonky there.

Reply:
It turns out I misread the value of `RUSTFLAGS` here, we're actually _disabling_ the `crt-static` feature, not enabling it (since that is the default, and would be represented as `+crt-static` if we were enabling it manually). This is needed in order to build dylibs on Alpine (which obviously we require for the extension). I've added a comment to that effect.

Reply:
I'm still hitting the same issue as before with your latest change. Perhaps it's an arm64 thing?

Reply:
That's strange, must not be related to the `RUSTFLAGS` then because it wasn't removed in my changes. I can't reproduce on x86_64 (I use a desktop machine in my house as my Docker builder, rather than my macbook). I'll try and do a build locally on my laptop and see if I can figure out what's going on.

Reply:
But your change does move `RUSTFLAGS` to be set _after_ pgx install.

Reply:
Ah, you know, I did move it down to above our `make package` invocation, perhaps it is needed by `cargo install pgx` or something. I'll move it up to above all the cargo builds.

Reply:
The latest commit fixes this 👍